packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js

```diff
@@ -70,4 +92,74 @@ export default class QuanticSmartSnippetSource extends LightningElement {
       );
     }
   }
+
+  /**
+   * Returns the smart snippet source title.
+   * @returns {string}
+   */
+  get sourceTitle() {
+    return this.source?.title;
+  }
+
+  /**
+   * Returns the smart snippet source uri.
+   * @returns {string}
+   */
+  get sourceUri() {
+    return this.source?.clickUri;
+  }
+
+  handleClick(event) {
+    if (this.isSalesforceLink) {
+      event.preventDefault();
+      this.navigateToSalesforceRecord(event);
+    }
+  }
+
+  navigateToSalesforceRecord(event) {
+    event.stopPropagation();
+    const targetPageRef = {
+      type: 'standard__recordPage',
+      attributes: {
+        recordId: this.recordIdAttribute,
+        actionName: 'view',
+      },
+    };
+    this[NavigationMixin.Navigate](targetPageRef);
+  }
+
+  get recordIdAttribute() {
+    // Knowledge article uses the knowledge article version id to navigate.
+    if (this.source?.raw?.sfkbid && this.source?.raw?.sfkavid) {
```

Comment:
I don't think you need `sfkbid` if you only ever use `sfkavid` ?

Reply:
Wherever we are dealing with knowledge articles we always start  by confirming that `sfkbid` exists before starting the work `sfkavid`, for example: https://github.com/coveo-platform/salesforceintegration/blob/30a2fdfd20d2b61ebca06519c2062ac245a527b8/force-app/2gp/hosted-experiences/lwc/utilsHostedExperiences/attachToCaseUtils.js#L46

I'm not sure why tbh, but it seems to me as a sanity check to make sure we are dealing with a valid knowledge cause there should be no situation where  `sfkavid` will be present and `sfkbid` will not.